specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json

```diff
@@ -2733,6 +2730,15 @@
             "required": true,
             "type": "string"
           },
+          {
+            "name": "validate",
```

Comment:
We have a single validate endpoint on service side but here we have 2 swagger definitions "Workflows_ValidateByResourceGroup" and "ValidateByLocation" both of which are calling into same validate endpoint but with different input parameters. AFAIK, "workflow definition to validate" is always expected as an input to the API. If the workflow already exists, validate ensures that incoming definition is a valid patch otherwise it validates the new workflow.

Workflows_ValidateByResourceGroup's contract was written correctly to accept workflow as input, but same was missing in Workflows_ValidateByLocation, hence this change.

Reply:
This is more API design question than syntax. I find it strange you have validate by Location and Resource group. With OData design, it would be more natural if you have two validate Actions. One on specific workflow entity and one on entity collection for generic validation. I don't see why location itself needs to factoring into the validate. You have not other API (CRUD) that directly address a workflow via location. 

Here is what I meant,    
 
POST  \resourceGroup\{RG}\provider\Microsoft.Logic\workflows\{workflow name}\validate    (This should NOT take any parameter as it is just validating an existing workflow, this API may not even be needed if you do validation during create or update that had ensured only valid workflow is accepted.  But I suppose that if you have later system schema update would call for re-validation then this may be some of use.}
POST \provider\Microsoft.Logic\workflows\validate    (This is an stateless function on entity collection which REQUIRES a workflow in the body to do ad-hoc validation. It is not tie to any existing workflow and can be used as an utility API for various consumer to use. Designer, End user etc.)




---
In reply to: [403373536](https://github.com/Azure/azure-rest-api-specs/pull/8934#discussion_r403373536) [](ancestors = 403373536)

Reply:
The main reason for use of location is to route the validation calls to particular region aware endpoint as as the workflow definition could have region specific parameters that need region aware validation. For example, our west us region will not have any knowledge of connections in east us and we need to make sure westus workflows are validation by the west us endpoint.

Reply:
I see. Then I am still baffled by the inclusion of the workflow itself in the request. Since you are already referring to a Workflow instance.

Reply:
Validation will always be performed on passed-in/incoming workflow. It is possible that workflow passed-in already exists, in which case we will still validate each and every property of incoming workflow to ensure that incoming workflow will still result in a valid workflow if followed by for e.g subsequent patch/PUT. 
Internally, PUT workflow also calls into same helper to perform validation on workflow prior to create/update. In case it helps, i can share a link to Validate API code on service side.